src/Compatibility/GenAPI/Microsoft.DotNet.GenAPI/CSharpAssemblyDocumentGenerator.cs

```diff
@@ -25,75 +25,45 @@ namespace Microsoft.DotNet.GenAPI;
-    /// <param name="metadataReferences">The metadata references to use. The default value is <see langword="null"/>.</param>
-    /// <param name="diagnosticOptions">The optional diagnostic options to use. The default value is <see langword="null"/>.</param>
-    /// <param name="addPartialModifier">Whether to add the partial modifier or not. The default value is <see langword="true"/>.</param>
-    /// <param name="hideImplicitDefaultConstructors">Whether to hide implicit default constructors or not. The default value is <see langword="true"/>.</param>
-    public CSharpAssemblyDocumentGenerator(ILog log,
-                                 IAssemblySymbolLoader loader,
-                                 ISymbolFilter symbolFilter,
-                                 ISymbolFilter attributeDataSymbolFilter,
-                                 string? exceptionMessage,
-                                 bool includeAssemblyAttributes,
-                                 IEnumerable<MetadataReference>? metadataReferences = null,
-                                 IEnumerable<KeyValuePair<string, ReportDiagnostic>>? diagnosticOptions = null,
-                                 bool addPartialModifier = true,
-                                 bool hideImplicitDefaultConstructors = true)
+    /// <param name="options">The options to configure the generator.</param>
+    public CSharpAssemblyDocumentGenerator(ILog log, CSharpAssemblyDocumentGeneratorOptions options)
     {
         _log = log;
-        _loader = loader;
-        _symbolFilter = symbolFilter;
-        _attributeDataSymbolFilter = attributeDataSymbolFilter;
-        _exceptionMessage = exceptionMessage;
-        _includeAssemblyAttributes = includeAssemblyAttributes;
+        _options = options;
+
         _adhocWorkspace = new AdhocWorkspace();
         _syntaxGenerator = SyntaxGenerator.GetGenerator(_adhocWorkspace, LanguageNames.CSharp);
-        _metadataReferences = metadataReferences;
-        _addPartialModifier = addPartialModifier;
-        _hideImplicitDefaultConstructors = hideImplicitDefaultConstructors;
 
         _compilationOptions = new CSharpCompilationOptions(
             OutputKind.DynamicallyLinkedLibrary,
             nullableContextOptions: NullableContextOptions.Enable,
-            specificDiagnosticOptions: diagnosticOptions);
+            specificDiagnosticOptions: _options.DiagnosticOptions);
     }
 
     /// <summary>
     /// Returns the configured source code document for the specified assembly symbol.
     /// </summary>
     /// <param name="assemblySymbol">The assembly symbol that represents the loaded assembly.</param>
     /// <returns>The source code document instance of the specified assembly symbol.</returns>
-    public Document GetDocumentForAssembly(IAssemblySymbol assemblySymbol)
+    public async Task<Document> GetDocumentForAssemblyAsync(IAssemblySymbol assemblySymbol)
```

Comment:
Something inside this method is still reducing the fully qualified names and removing namespace prefixes like `System.` or `Microsoft.`. I already set the `ReducerAsync` optional and removed the `Simplifier.Annotation` instance from `AdditionalAnnotations` list. See https://github.com/dotnet/sdk/pull/46425/files#r1984035962